core/state_processor.go

```diff
@@ -161,7 +161,8 @@ func ApplyTransactionWithEVM(msg *Message, gp *GasPool, statedb *state.StateDB,
 		statedb.AccessEvents().Merge(evm.AccessEvents)
 	}
 
-	return MakeReceipt(evm, result, statedb, blockNumber, blockHash, tx, *usedGas, root), nil
```

Comment:
Are you sure? The `receipt` variable will be assigned automatically by the return value, isn't it?

```go
func foo() (result bool) {
	defer func() {
		fmt.Println("result", result)
	}()
	return true
}

func TestFoo(t *testing.T) {
	foo()
}
```
This script prints true for me

Reply:
Great catch @rjl493456442. 
I am working on a tracer and I was receiving nil value for the receipt on **OnTxEnd**. 
Let me check it again on the go-ethereum fork I work on, as it might be that something else I did which fixed it, and I wrongly thought that this was the fix.

I will update or close the issue today based on my finding. Thanks

Reply:
We can close it. The bug was somewhere else. Sorry for the false PR.